### PR TITLE
For API attachments, use API URL

### LINKED ISSUE #25257: Cannot access private repo release files through API using bearer token anymore
### Description

Apparently an RC release was pushed to Docker instances using the "lates" tag, so my Gitea instance is now running 1.20.0rc and I cannot downgrade. Since updating, I am no longer able to download files over HTTPS using my token. I am attempting to access the file URL and I am getting 404 errors. If I make the repository public, I am able to download the files just fine.

This bearer token works when listing releases via the API, just not downloading them. Nothing else has changed in my instance.

### Gitea Version

1.20.0+rc0-48-g3afc3e4a7

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Docker

### Database

MySQL

### PR DESCRIPTION
**Pull request by lunny**
Fix #25257

### CODE REVIEW

**Comment by JakobDev** [comment]
Does this revert #23891 on the API side?

**Comment by lunny** [comment]
> Does this revert #23891 on the API side?

It will not. It will change the API related release attachments' URL. But I think #23891 hasn't touched any API returning.

**Comment by JakobDev** [comment]
It has touched the API. The release API also returns the new URL. If you revert this, we will end up with the old `/attachment/<uuid>` when a API is involved. So if you use e.g. an automatic Update checker, on multiple Repos you will end up with multiple `/attachment/<uuid>` URLs. If you later want to debug them, then good luck.  You will need to download each file to see what it is. With the new URL, you can just tell by locking at the URL which Version from which repo you will get.

**Comment by lunny** [comment]
Or we can add a new user-friendly endpoint for API.

**Comment by lunny** [comment]
> 

Because those are for different aims. Web requests will check users' session, but API will check tokens. Especially for private repositories, they are different. But we can share the functions even if endpoints are different.

**Comment by JakobDev** [comment]
This function is responsible fr serving the Attachment:
https://github.com/go-gitea/gitea/blob/45bc180a15d1edf3a20f25c4bb428819b94e6ad9/routers/web/repo/attachment.go#L89-L152
Maybe you can add the check here. `/attachment/<uuid>` needs to keep working for backward compatibility, so we need to fix this function anyway.

**Comment by lunny** [comment]
> This function is responsible fr serving the Attachment:
> 
> https://github.com/go-gitea/gitea/blob/45bc180a15d1edf3a20f25c4bb428819b94e6ad9/routers/web/repo/attachment.go#L89-L152
> 
> Maybe you can add the check here. `/attachment/<uuid>` needs to keep working for backward compatibility, so we need to fix this function anyway.

This endpoint will not be affected by this PR.

**Review by silverwind — APPROVED** [review]

File: routers/api/v1/repo/release_attachment.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 
 // GetReleaseAttachment gets a single attachment of the release
 func GetReleaseAttachment(ctx *context.APIContext) {
-	// swagger:operation GET /repos/{owner}/{repo}/releases/{id}/assets/{attachment_id} repository repoGetReleaseAttachment
+	// swagger:operation GET /repos/{owner}/{repo}/releases/{id}/assets/{asset} repository repoGetReleaseAttachment
```
Comment (wxiaoguang) [inline routers/api/v1/repo/release_attachment.go]: I still think attachment_id is right here.

1. It was documented as so, it's a bug in code so no client should ever work.
2. It is an ID (integer)
3. The code is doing `GetAttachmentByID(ctx, attachID)`

Comment (lunny) [reply routers/api/v1/repo/release_attachment.go]: I think there is an inconsistent in the previous code. the router in api.go use the parameter variable `asset` but the document using `attachment_id`. If you think change all the places to `attachment_id` is better, I can do that.

Comment (wxiaoguang) [reply routers/api/v1/repo/release_attachment.go]: Personally I prefer "attachment_id" everywhere, it's more consistent, it won't break the legacy code or generated client API code.

If it's changed to "asset", the API itself changes, the generated API client code would also change?

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**